### PR TITLE
workflow dependencies graph visualisation

### PR DESCRIPTION
**Pull request by itayL1**
add support for the visualization of the workflow dependencies graph of nested workflow templates (in a new isolated package). 

example usage:
for this template (from the test):
image

this graph was generated:
image

### CODE REVIEW

File: argo_workflow_tools_visualization/argo_workflow_tools_visualization/dsl/dependencies_graph/build_workflow_dependencies_graph.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        self._graph: nx.DiGraph = None
+
+    def build_workflow_graph(self, compiled_workflow):
+        self._graph = nx.DiGraph()
+
+        self.add_template_subgraph(compiled_workflow, parent_node_id=None)
+        return self._graph
+
+    def add_template_subgraph(self, dsl_wf_template: WorkflowTemplate, parent_node_id: Optional[str]):
+        wf_template = dsl_wf_template.to_model(embed_workflow_templates=False)
```
[the remark this answers is not in the document]
Comment (itayL1) [reply argo_workflow_tools_visualization/argo_workflow_tools_visualization/dsl/dependencies_graph/build_workflow_dependencies_graph.py:34]: my mistake, it doesn't work when embed_workflow_templates=True, as it should

**Comment by mzedayda** [comment]
In the current implementation, the graph only shows parent-child dependencies, is that what we want? wouldn't it be more informative to print the actual dag?

File: Makefile [hunk 2]
```diff
@@ -8,7 +8,7 @@ isort:
 	isort argo_workflow_tools tests examples
 
 test:
-	poetry run pytest --durations=5 tests/dsl
+	poetry run pytest --durations=5 tests/argo_workflow_tools/dsl/
```
[the remark this answers is not in the document]
Comment (itayL1) [reply Makefile:11]: same as it was before. also, the tests in tests/argo_workflow_tools/client don't work.

File: argo_workflow_tools_visualization/pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+[tool.poetry]
```
[the remark this answers is not in the document]
Comment (itayL1) [reply argo_workflow_tools_visualization/pyproject.toml:1]: yes, it's a bit weird, but I don't think that it is worth another repo, which will be less convenient. I was actually surprised that poetry doesn't support multiple packages under the same project ([here](https://github.com/python-poetry/poetry/issues/161) is a thread on it). WDYT?

**Comment by itayL1** [comment]
@mzedayda Yes what I want is the high-level dependencies graph between the workflows. this is very useful at least in our project, where we use nest many workflows in the complicated release flows. also, plotting the real dag is very complex, there are a lot of edge cases that are resolved by argo at runtime (loops, condition etc.).

**Pull request merged**